### PR TITLE
Automatic Login Refreshing

### PR DESCRIPTION
**Pull request by Dala0**
## What does this change?
Adds code which attempts to refresh the login data two minutes before it expires. The code is heavily based on existing code from pluto-headers.

## How to test
The software should refresh the login data two minutes before it expires. You should see 'Login refreshed' in the console after a successful refresh.

## How can we measure success?
The login data gets refreshed.

This was tested on a machine running macOS 10.12.6 and on three virtual machines.

@fredex42

### CODE REVIEW

File: conf/application.conf [hunk 1]
```diff
@@ -131,6 +131,7 @@ pluto {
 
 auth {
   tokenSigningCertPath = "keycloak-local.pem"
+  adminClaimName = "multimedia_admin"
```
Comment (fredex42) [inline conf/application.conf:134]: I'd put a placeholder in here for commit to the repo if i were you 😄

File: frontend/app/LoginRefreshComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    /**
+     * Called periodically every second once a refresh has failed to alert the user how long they have left
+     */
+    const updateCountdownHandler = () => {
+        const nowTime = new Date().getTime() / 1000; //assume time is in seconds
+        const expiry = loginDataRef.current.exp;
+        const timeToGo = expiry - nowTime;
+
+        if(timeToGo>1) {
```
Comment (fredex42) [inline frontend/app/LoginRefreshComponent.tsx:57]: could be simplified to `if(timeToGo<=1 && props.onLoginExpired) props.onLoginExpired();`

**Review by fredex42 — CHANGES_REQUESTED** [review]
Looks good, just a couple of very minor points

File: conf/application.conf (hunk 1 above)
Comment (Dala0) [reply conf/application.conf]: Done.

File: frontend/app/LoginRefreshComponent.tsx (hunk 2 above)
Comment (Dala0) [reply frontend/app/LoginRefreshComponent.tsx]: Done.

**Review by fredex42 — APPROVED** [review]
👍

**Pull request merged**